frontend/tests/utils/wmgUtils.ts

```diff
@@ -145,3 +147,235 @@ export const checkPlotSize = async (page: Page) => {
+
+  // Save downloaded file in a directory
+  await download.saveAs(zipFilePath);
+
+  //extract zip file
+  const zip = new AdmZip(zipFilePath);
+  zip.extractAllTo(extractDirPath);
+};
+
+export const getFilterText = async (page: Page, filterName: string) => {
+  const filter_label = `${getTestID(filterName)} [role="button"]`;
+  return await page.locator(filter_label).textContent();
+};
+export const verifyMetadata = async (
+  page: Page,
+  filterName: string,
+  data: string[],
+  noSelectionText = "No selection"
+) => {
+  //verify the date is valid
+  const dateString = data[0].substring(14);
+  const date = new Date(dateString);
+
+  // Check if the resulting date is valid
+  expect(!isNaN(date.getTime())).toBe(true);
+
+  expect(
+    data[1].includes(
+      "# We regularly expand our single cell data corpus to improve results. Downloaded data and figures may differ in the future."
+    )
+  ).toBeTruthy();
+
+  //check if the 3 column contains the gene expression link
+  expect(
+    data[2].includes("https://localhost:3000/gene-expression")
+  ).toBeTruthy();
+
+  // Extract the link using a regular expression
+  const linkRegex = /https?:\/\/[^\s]+/;
+  const linkMatch = data[2].match(linkRegex);
+  let text: string | null = "";
+  if (filterName !== "no-filter") {
+    text = await getFilterText(page, filterName);
+  }
+
+  // Check if a match was found and log the result
+  if (linkMatch) {
+    const link = linkMatch[0];
+    //verify link is valid
+
+    await page.goto(link);
+
+    // wait until the new page fully loads
+    await page.waitForLoadState();
+    // expect the header on the new page to be visible
+    expect(page.getByTestId("download-button")).toBeVisible();
+  }
+
+  //verify the text displayed in the filter section of meta data
+  switch (filterName) {
+    case "dataset-filter":
+      expect(
+        data[3].includes("# Dataset Filter Values: " + `${text}`)
+      ).toBeTruthy();
+      expect(
+        data[4].includes(`# Disease Filter Values: ${noSelectionText}`)
+      ).toBeTruthy();
+      expect(
+        data[5].includes(
+          `# Self-Reported Ethnicity Filter Values: ${noSelectionText}`
+        )
+      ).toBeTruthy();
+      expect(
+        data[6].includes(`# Sex Filter Values: ${noSelectionText}`)
+      ).toBeTruthy();
+      break;
+    case "disease-filter":
+      expect(data[4].includes(`# Disease Filter Values: ${text}`)).toBeTruthy();
+      expect(
+        data[3].includes(`# Dataset Filter Values: ${noSelectionText}`)
+      ).toBeTruthy();
+      expect(
+        data[5].includes(
+          `# Self-Reported Ethnicity Filter Values: ${noSelectionText}`
+        )
+      ).toBeTruthy();
+      expect(
+        data[6].includes(`# Sex Filter Values: ${noSelectionText}`)
+      ).toBeTruthy();
+      break;
+    case "self-reported-ethnicity-filter":
+      expect(
+        data[5].includes(`# Self-Reported Ethnicity Filter Values: ${text}`)
```

Comment:
Extract this logic into a generic function that takes data[x] and filter value as parameters. This will make this function much more simpler and easier to maintain